primitives/api/proc-macro/src/decl_runtime_apis.rs

```diff
@@ -188,13 +189,17 @@ fn generate_runtime_decls(decls: &[ItemTrait]) -> Result<TokenStream> {
 		let mut decl = decl.clone();
 		let decl_span = decl.span();
 		extend_generics_with_block(&mut decl.generics);
-		let metadata = generate_decl_runtime_metadata(&decl);
 		let mod_name = generate_runtime_mod_name_for_trait(&decl.ident);
 		let found_attributes = remove_supported_attributes(&mut decl.attrs);
 		let api_version =
 			get_api_version(&found_attributes).map(|v| generate_runtime_api_version(v as u32))?;
 		let id = generate_runtime_api_id(&decl.ident.to_string());
 
+		#[cfg(feature = "frame-metadata")]
+		let metadata = generate_decl_runtime_metadata(&decl);
```

Comment:
```suggestion
		let metadata = crate::runtime_metadata::generate_decl_runtime_metadata(&decl);
```

Then you can remove

```rust
#[cfg(feature = "frame-metadata")]
use crate::runtime_metadata::generate_decl_runtime_metadata;
```

altogether and there will be only one `#[cfg(feature = "frame-metadata")]` necessary instead of two. (Same for the other file.)